File: sg_cdez_api/src/main/java/com/cdez/sg_cdez_api/dto/response/PersonalResponse.java
package com.cdez.sg_cdez_api.dto.response;

import com.cdez.sg_cdez_api.entity.enums.Especialidad;
import com.cdez.sg_cdez_api.entity.enums.TipoIdentificacion;

import java.time.LocalDateTime;
import java.util.*;

public record PersonalResponse(
        UUID personalId,
        RolResponse rol,
        Especialidad especialidad,
        TipoIdentificacion tipoIdentificacion,
        String identificacion,
        String primerNombre,
        String segundoNombre,
        String primerApellido,
        String segundoApellido,
        String direccion,
        String carnet,
        String usuario,
        String activo,
        String createdBy,
        LocalDateTime createdAt,
        String updatedBy,
        LocalDateTime updatedAt,

        List<ContactoResponse> contactos,
        List<DocumentoResponse> documentos
) {
}

File: sg_cdez_web/src/services/interfaces/personalResponse.ts
export interface ContactoResponse {
    contactoId: number;
    personalNombre: string;
    encargadoNombre: string;
    valor: string;
    tipoValor: string;
    activo: string;
    createdBy: string;
    createdAt: string;
    updatedBy: string;
    updatedAt: string;
}

export interface DocumentoResponse {
    documentoId: number | null;
    adultoId: string;
    nombreArchivo: string;
    tipoArchivo: string;
    tamanoArchivo: number;
    activo: string;
    createdAt: string;
    updatedAt: string;
}

export interface RolResponse {
    id: number;
    nombre: string;
}

export interface PersonalResponse {
    personalId: string;
    rol: RolResponse;
    especialidad: string;
    tipoIdentificacion: string;
    identificacion: string;
    primerNombre: string;
    segundoNombre: string;
    primerApellido: string;
    segundoApellido: string;
    direccion: string;
    carnet: string;
    usuario: string;
    activo: string;
    createdBy: string;
    createdAt: string;
    updatedBy: string;
    updatedAt: string;

    contactos: ContactoResponse[];
    documentos: DocumentoResponse[];
}